Content.Server/Nutrition/EntitySystems/DrinkSystem.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Content.Server.Nutrition.Components;
+using Content.Shared.Chemistry.Components.SolutionManager;
+using Content.Shared.Chemistry.EntitySystems;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+
+namespace Content.Server.Nutrition.EntitySystems
+{
+    [UsedImplicitly]
+    public class DrinkSystem : EntitySystem
+    {
+        [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<DrinkComponent, SolutionChangedEvent>(OnSolutionChange);
+            SubscribeLocalEvent<DrinkComponent, ComponentInit>(OnDrinkInit);
+        }
+
+        private void OnDrinkInit(EntityUid uid, DrinkComponent component, ComponentInit args)
+        {
+            component.Opened = component.DefaultToOpened;
+
+            var owner = _entityManager.GetEntity(uid);
+            if (owner.TryGetComponent(out DrainableSolutionComponent? existingDrainable))
```

Comment:
Just use ComponentManager directly and avoid going into EntityManager, this is also in a few spots.